src/PrintReceipt.ts

```diff
@@ -1,12 +1,179 @@
+/* eslint-disable no-trailing-spaces */
+/* eslint-disable no-var */
+/* eslint-disable indent */
+/* eslint-disable prefer-const */
 import {loadAllItems, loadPromotions} from './Dependencies'
 
+interface ReceiptPrintingInfo {
+  name: string;
+  quantity: string;
+  unitPrice: string;
+  subTotal: string;
+  unit: string
+  discountedPrice: string
+}
+
+interface ReceiptItem {
+  name: string;
+  quantity: number;
+  unit: string
+  unitPrice: number;
+  subtotal: number;
+  discountedPrice: number;
+  ifPromoted: boolean;
+  barcode: string;
+}
+
+interface Tag {
+  barcode: string;
+  quantity: number;
+}
+
 export function printReceipt(tags: string[]): string {
-  return `***<store earning no money>Receipt ***
-Name：Sprite，Quantity：5 bottles，Unit：3.00(yuan)，Subtotal：12.00(yuan)
-Name：Litchi，Quantity：2.5 pounds，Unit：15.00(yuan)，Subtotal：37.50(yuan)
-Name：Instant Noodles，Quantity：3 bags，Unit：4.50(yuan)，Subtotal：9.00(yuan)
-----------------------
-Total：58.50(yuan)
-Discounted prices：7.50(yuan)
-**********************`
+    let parsedTags = parseTags(tags)
+    let productsInfos = getProductsInfo(parsedTags)
+    checkIfPromoted(productsInfos)
+    calculatePrice(productsInfos)
+    let printingInfos = getReceiptPritingInfo(productsInfos)
+    return constructReceipt(printingInfos)
+}
+
+function parseTags(tags: string[]): Tag[] {
+  let map = new Map<string, number>()
+  let parsedTags: Tag[] = []
+
+  for (let tag of tags) {
+    let splitedTag = tag.split('-')
+    let barcode = splitedTag[0]
+    let quantity = 1
+    if (splitedTag.length > 1) {
+      quantity = Number(splitedTag[1])
+      console.log(quantity)
+    }
+    
+
+    if (!checkIfBarcodeExist(barcode)) {
```

Comment:
Probably validation part could be extracted into an independent function to make code more readable.